week3/feedback/feedback_server/database.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"errors"
+	pb "grab/week3/feedback/feedback"
+)
+
+type Feedbacks struct {
+	ExistingBookingCode map[string]bool
+	ExistingPassengerID map[int32]bool
+	Feedbacks           []pb.PassengerFeedback
+}
+
+// CreateNewFeedbacks create new feedback database
+func CreateNewFeedbacks() Feedbacks {
+	return Feedbacks{
+		ExistingBookingCode: make(map[string]bool),
+		ExistingPassengerID: make(map[int32]bool),
+		Feedbacks:           make([]pb.PassengerFeedback, 0),
+	}
+}
+
+// AddFeedback add new feedback to database
+func (fb *Feedbacks) AddFeedback(feedback *pb.PassengerFeedback) error {
+	if fb.ExistingBookingCode[feedback.BookingCode] == true {
+		return errors.New("Duplicated booking code")
+	}
+	fb.ExistingBookingCode[feedback.BookingCode] = true
+	fb.ExistingPassengerID[feedback.PassengerID] = true
+	fb.Feedbacks = append(fb.Feedbacks, *feedback)
+	return nil
+}
+
+// DeleteFeedbacksByPassengerID delete all feedbacks by passengerID
+func (fb *Feedbacks) DeleteFeedbacksByPassengerID(psgID int32) (int32, error) {
+	if fb.ExistingPassengerID[psgID] == false {
+		return 0, errors.New("Non-existing passengerId")
+	}
+	var newFeedbacks []pb.PassengerFeedback
+	var cntDeleted int32
+	for _, feedback := range fb.Feedbacks {
```

Comment:
looping through all of feedbacks and append it to new array has very poor performance
simply using `Feedbacks  map[string]pb.PassengerFeedback` will help a lots